### PR TITLE
Update CHANGELOG for Firestore v1.16.4

### PR DESCRIPTION
**Pull request by wilhuff**

### CODE REVIEW

**Comment by wilhuff** [comment]
I note that the release spreadsheet has this change down as 1.17.0 though there's no change that would seem to warrant that. Does this need to be 1.17.0?

File: Firestore/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Unreleased
 
+# v1.16.4
```
Comment (paulb777) [inline Firestore/CHANGELOG.md:3]: I'd like to be conservative and bump to 1.17.0 for more flexibility

Comment (wilhuff) [reply Firestore/CHANGELOG.md:3]: What does "more flexibility" mean exactly? The downside of bumping is that this is the end of patch releases that can be trivially picked up by the C++ SDK that builds on top of this.

**Comment by paulb777** [comment]
We're making a change that is not a fix to facilitate a future feature - so it's not really a patch.

File: Firestore/CHANGELOG.md (hunk 1 above)
Comment (wilhuff) [reply Firestore/CHANGELOG.md:3]: I should clarify: C++ releases are going to be more infrequent than iOS ones so I'd like preserve the flexibility to release fixes there to those users.

Meanwhile, this change is functionally transparent to iOS users. It's not an addition or a removal--and it doesn't matter what our intent is.

Comment (paulb777) [reply Firestore/CHANGELOG.md:3]: OK, it should be fine to leave as a patch for Firestore.  I feel more strongly about pods with intra-repo dependencies for flexibility in being able to patch if something does go wrong and minimizing the version exponential surface area with different file structures.

**Pull request merged**